src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs

```diff
@@ -235,6 +237,27 @@ public static Encoding GetEncoding(int codepage)
                 case CodePageNoThread:                           // 3 CP_THREAD_ACP
                 case CodePageNoSymbol:                           // 42 CP_SYMBOL
                     throw new ArgumentException(SR.Format(SR.Argument_CodepageNotSupported, codepage), nameof(codepage));
+
+                case CodePageUTF7:                               // 65000
+                    {
+                        // Support for UTF-7 is disabled by default. It can be re-enabled by registering a custom
+                        // provider (which early-exits this method before the 'switch' statement) or by using
```

Comment:
What custom provider?

Reply:
If you call [`Encoding.RegisterProvider`](https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding.registerprovider), you can override the return value of `Encoding.GetEncoding("utf-7")`, `Encoding.GetEncoding("utf-8")`, and any other encoding instance. If you register a custom provider for UTF-7, we never query the `AppContext` switch because this switch statement never get hit.

Registering a custom provider doesn't change the return value of the built-in accelerators (like `Encoding.UTF7` or `Encoding.UTF8`) since they're hardcoded to always use the built-in implementation.

Reply:
Right.  It's the "re-enabled" lingo that confuses me.  There's no way for someone to get at our built-in Encoding to re-enable it via Encoding.RegisterProvider, is there?